index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/client/EsDocument.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr.index.elasticsearch.client;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.infinispan.schematic.DocumentFactory;
+import org.infinispan.schematic.document.Document;
+import org.infinispan.schematic.document.EditableDocument;
+import org.infinispan.schematic.document.Json;
+import org.infinispan.schematic.internal.document.BasicArray;
+
+/**
+ * Json document used for exchange data with Elasticsearch engine.
+ * 
+ * @author kulikov
+ */
+public class EsDocument {
```

Comment:
I'm curious why this class even exists. It seems like it is less efficient in most read-only cases, since this class wraps an _editable_ document, which may not be needed when the EsDocument is just read.